### PR TITLE
[Cherry-pick]Add DistributedBatchSampler and Colerjitter

### PR DESCRIPTION
**Pull request by LielinJiang**
### PR types

New features

### PR changes

APIs
### Describe

Add DistributedBatchSampler and Colerjitter for 1.8

image
image

### CODE REVIEW

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by swtkiwi — APPROVED** [review]
LGTM

File: python/paddle/incubate/hapi/vision/transforms/transforms.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        with np.errstate(over="ignore"):
+            h += np.uint8(alpha * 255)
+        hsv_img = cv2.merge([h, s, v])
+        return cv2.cvtColor(hsv_img, cv2.COLOR_HSV2BGR_FULL).astype(dtype)
+
+
+class ColorJitter(object):
+    """Randomly change the brightness, contrast, saturation and hue of an image.
+    Args:
+        brightness: How much to jitter brightness.
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/vision/transforms/transforms.py:182]: 建议修改：brightness: How much to jitter brightness. The value, as a non-negative number, is generated uniformly from [max(0, 1 - brightness), 1 + brightness]or from the given [min, max].

File: python/paddle/incubate/hapi/distributed.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    is exclusive to it.
+    .. note::
+        Dataset is assumed to be of constant size.
+        
+    Args:
+        dataset(paddle.io.Dataset): this could be a `paddle.io.Dataset` implement
+                     or other python object which implemented
+                     `__len__` for BatchSampler to get sample
+                     number of data source.
+        batch_size(int): sample indice number in a mini-batch indices.
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/distributed.py:41]: 建议修改：The number of samples in a mini-batch index.

File: python/paddle/incubate/hapi/distributed.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        shuffle(bool): whther to shuffle indices order before genrating
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/distributed.py:42]: whether... generating...

File: python/paddle/incubate/hapi/distributed.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            batch indices. Default False.
+        drop_last(bool): whether drop the last incomplete batch dataset size
+            is not divisible by the batch size. Default False
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/distributed.py:45]: whether to drop the last incomplete batch dataset which is not divisible by BatchSampler. Default False.

File: python/paddle/incubate/hapi/distributed.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    as a DataLoader sampler, and load a subset of the original dataset that 
+    is exclusive to it.
+    .. note::
+        Dataset is assumed to be of constant size.
+        
+    Args:
+        dataset(paddle.io.Dataset): this could be a `paddle.io.Dataset` implement
+                     or other python object which implemented
+                     `__len__` for BatchSampler to get sample
+                     number of data source.
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/distributed.py:40]: a `paddle.io.Dataset` implementation or other python object is implemented by `__len__`  to calculate the sample number of the data source used in BatchSampler.不确定是否改变了原意

File: python/paddle/incubate/hapi/distributed.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+from paddle.fluid.dygraph.parallel import ParallelEnv
+from paddle.fluid.io import BatchSampler
+
+__all__ = ['DistributedBatchSampler']
+
+
+class DistributedBatchSampler(BatchSampler):
+    """Sampler that restricts data loading to a subset of the dataset.
+    In such case, each process can pass a DistributedBatchSampler instance
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/distributed.py:30]: In such cases

**Pull request merged**